### ISSUE TITLE
`no-unused-vars` false negative with comma operator

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#14866]
**Tell us about your environment**

* **ESLint Version:** 7.30.0
* **Node Version:** v14.x
* **npm Version:**  v6.x
* **Operating System:**  MacOS

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
/*eslint no-unused-vars: "error"*/
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/*eslint no-unused-vars: "error"*/

var z = 0;
z=z+1, z=2;

var x = 0;
x++, x = 2;

var y = 0;
y = y+1;
```

**What did you expect to happen?**

`no-unused-vars` should report all`x`, `y`, and `z`  as unused variables.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

`no-unused-vars` is reporting only `x`  and `y` as unused variables.

**Steps to reproduce this issue:**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFwiZXJyb3JcIiovXG5cbnZhciB6ID0gMDtcbno9eisxLCB6PTI7XG5cbnZhciB4ID0gMDtcbngrKywgeCA9IDI7XG5cbnZhciB5ID0gMDtcbnkgPSB5KzE7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

We probably missed addressing this case in https://github.com/eslint/eslint/issues/14325

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by nzakas** [comment]
Is this related to https://github.com/eslint/eslint/issues/14325?

**Comment by snitin315** [comment]
> Is this related to #14325?

@nzakas It is similar to #14325,  just `x=x+1` instead of `x++`.

**Comment by nzakas** [comment]
Got it.

**Comment by nzakas** [comment]
Since this is kind of a continuation of the previous issue, marking as accepted.

**Comment by ersachin3112** [comment]
can i work on this issue?

**Comment by snitin315** [comment]
@ersachin3112 Sure, let us know if you need any help.
